### ISSUE TITLE
Coverage report missing for cancer scout

### ISSUE THREAD

**Issue by heronikdin** [Clinical-Genomics/scout#2253]
Hi (from cust087, do I have to write the cust no everytime?) 

We have seen from the 2 first myeloid samples that we have analysed in cancer scout, that the quality metrics are missing from the pdf report. How do we conclude that the data presenterd is approved technically from a coverage point of view?
According to Bianca Tesi, for GMCK panel the coverage of genes is represented in the PDF file, and for the other panels/runs there is a summarized coverage for the specific panel/run. 

Hälsningar Hero

**Comment by hassanfa** [comment]
I can check if something can be done on BALSAMIC side.  In BALSAMIC we produce all those metrics, but I'm not sure how to propagate them to Scout.

Maybe @dnil knows: How does the coverage report look like for GMCK panel in non-cancer cases? And how to produce it? What does Scout need? In BALSAMIC we have coverage from mosdepth tool, and callable regions available.

**Comment by moonso** [comment]
How about passing on the multiqc-html report? Would that cover the needs?

**Comment by moonso** [comment]
If it is only about coverage we should utilise chanjo and produce chanjo reports for cancer cases as well

**Comment by dnil** [comment]
Or just run chanjo?

**Comment by dnil** [comment]
The delivery reports for the case would probably also get you far: I guess if BALSAMIC does not run chanjo, it will not get the overall coverage in them from CG either?

**Comment by henrikstranneheim** [comment]
No that is correct. And there is no cg infrastructure for mosdepth so far

**Comment by dnil** [comment]
Thanks @heronikdin - and nope, no need to mention your customer number unless you have issues that are possibly specific to your account. In which case you should probably contact the Solna site supportsystem directly instead (see https://clinical.scilifelab.se or email supportsystem@clinicalgenomics.se)!

**Comment by dnil** [comment]
mosdepth looks fairly pretty on the surface at least. A little more memory hungry than Sambamba, but faster to compensate that. I wonder how hard it would be to integrate in chanjo?

**Comment by dnil** [comment]
Actually we should both switch to samtools again.. :wink:
![Screenshot 2020-12-10 at 16 12 24](https://user-images.githubusercontent.com/758570/101790542-dede0e00-3b02-11eb-8959-e8991272f60e.png)
(From https://academic.oup.com/bioinformatics/article/34/5/867/4583630)

**Comment by hassanfa** [comment]
Sambamba is buggy and has problems with bedfiles. I stopped using sambamba longtime ago.

Multieqc should be uploaded to scout, and it has mosdepth coverage. And very neat plot.

But can someone please tell me what does chanjo provides that mosdepth can't? Seems like double work to use an external heavy designed tool to just read coverage data.

I will replicate a coverage report that chanjo produces, I just need specifications.

**Comment by hassanfa** [comment]
> Actually we should both switch to samtools again.. 😉
> ![Screenshot 2020-12-10 at 16 12 24](https://user-images.githubusercontent.com/758570/101790542-dede0e00-3b02-11eb-8959-e8991272f60e.png)
> (From https://academic.oup.com/bioinformatics/article/34/5/867/4583630)

RAM is cheap, we just allocate more. 😅

**Comment by hassanfa** [comment]
> No that is correct. And there is no cg infrastructure for mosdepth so far

I don't think we need a infrastructure to process a tsv file. :-)

**Comment by dnil** [comment]
> But can someone please tell me what does chanjo provides that mosdepth can't? Seems like double work to use an external heavy designed tool to just read coverage data.

Since one of our masters students made it, and the present solutions we have use it? :wink: 

But if there are issues with Sambamba (that we currently use) we should definitely consider replacing it. Perhaps make an issue on chanjo (or Sambamba), specifying the problems, and we see if or when they can be solved?

The real advantage with loading to chanjo is that the data goes into an aggregate database, and can be queried online later, so we can generate e.g. custom coverage reports for HPO panel searches on the fly via Chanjo-report in Scout.

Anyway, this is a sample coverage report:
![Screenshot 2020-12-10 at 16 29 36](https://user-images.githubusercontent.com/758570/101792707-4f862a00-3b05-11eb-8994-a8bb4ebb7782.png)

This can be interactively customised since its a Chanjo-report, but I think Hero would be very happy also with a static html or pdf delivery report with coverage% at a few set coverage levels. For panels I imagine the current fixed fold coverage levels would be rather low. Say 30x, 90x, 150x, 1500x, 3000x, 6000x instead? Or @heronikdin, any levels you would you like to see? 

It is similar on the static delivery reports (html) from cg. Why not have a look in Scout on an RD case yourself: I couldn't find a delivery report for a demo case off the bat.

**Comment by hassanfa** [comment]
I reported that to sambamba 2 years ago, and no movement. It is written in D  😄 , needs a brave souls and a lot of courage.

For multiple coverage levels I made a simple PR to Picard, and now we can use that for variable levels of coverage https://github.com/broadinstitute/picard/pull/1542 ( i think latest release has it)

Thanks for the report example, I'll see what I can do. (While we do necessary chanjo changes)

**Comment by dnil** [comment]
It may or may not be difficult to change things up in chanjo - some parts are more inspired than other.

It should be simple enough to get it to run mosdepth:
compare the current CLI call: https://github.com/Clinical-Genomics/chanjo/blob/master/chanjo/sambamba.py
But if we shouldn't rerun mosdepth, the real kicker is getting the transcript/exon bed files in sync between BALSAMIC/mosdepth and Chanjo. Currently I suppose cg does this, by exporting exon/transcript definitions from Scout and running chanjo with those. Couldn't we just let CG do that also for the BALSAMIC cases? That seems like the least amount of work. And unless you strongly depend on the mosdepth values you could disable it in BALSAMIC to save those perhaps-not-so-cheap cpu cycles?

**Comment by hassanfa** [comment]
I check callable regions for mosdepth regularly, but it doesn't end up in customer deliveries. So it is part of balsamic anyway.

I don't know what is the design philosophy for cg, but it shouldn't run a complex analysis.

Doesn't MIP run chanjo?  When chanjo is fixed , balsamic can also run it for panel data.

**Comment by dnil** [comment]
The trickiness is in the definition files, as transcripts and exons change with new gene model releases once you leave fixed capture - and actually do for fixed captures as well, only some parts may be missing if knowledge of the gene changed after the capture was designed.

Currently cg does this, but this is certainly not a law of nature. The pipeline would need knowledge of the transcript definition and location + credentials on the database serving chanjo, but cg could of course pass it to the pipe instead of running it on upload. To me it would seem easier just to adopt the same procedure for BALSAMIC, but just have a quick chat with e.g. @moonso and @henrikstranneheim or @jemten to decide where to run it.

When you open that issue on chanjo with an example of what was not working with Sambamba we are one step closer changing to mosdepth (or samtools 😜)!

**Comment by dnil** [comment]
Oh, and I think it is slowly fine to let cg handle more things again, after the nice things done by @moonso and @Mropat as long as we are careful not to import other components code into it, only APIs - or just use versioned REST or CLI APIs. While cg should not do advanced analysis, it can definitely start it and propagate its results to other components.

**Comment by hassanfa** [comment]
Maybe we should move this to discussions on GitHub ;-), and I see your point. You be the judge :-D

**Comment by dnil** [comment]
Yes, or Slack directly. A general Discussions on the Clinical-Genomics level, or on e.g. the CG repo might be a good place for things like this that may or may not touch many packages, and/or their integration. 😸
